terminator/lib/transaction-monitor-workflow.js

```diff
@@ -0,0 +1,62 @@
+/** @import {WorkflowEvent, WorkflowStep} from 'cloudflare:workers' */
+import { WorkflowEntrypoint } from 'cloudflare:workers'
+import { getChainClient as defaultGetChainClient } from './chain.js'
+
+/**
+ * Workflow that monitors a transaction and starts cancel workflow if it takes
+ * too long
+ */
+export class TransactionMonitorWorkflow extends WorkflowEntrypoint {
+  /**
+   * @param {WorkflowEvent} event
+   * @param {WorkflowStep} step
+   */
+  async run(
+    { payload: { transactionHash } },
+    step,
+    { getChainClient = defaultGetChainClient } = {},
+  ) {
+    try {
+      // Wait for transaction receipt with timeout
+      await step.do(
+        `wait for transaction receipt ${transactionHash}`,
+        {
+          timeout: `5 minutes`,
+          retries: {
+            limit: 3,
+          },
+        },
+        async () => {
+          const { publicClient } = getChainClient(this.env)
+          return await publicClient.waitForTransactionReceipt({
+            hash: transactionHash,
+          })
+        },
+      )
+
+      await step.do(
+        'clean terminate_service_tx_hash when transaction is confirmed',
```

Comment:
To me it was surprising that we clear the hash after confirmation. Could we either keep it (for our record keeping) or rename the column to `pending_terminate_service_tx_hash`?

Reply:
My preference would be to keep it.